accounts/views.py

```diff
@@ -101,13 +115,185 @@ def signup(request):
         return render(request, "accounts/login.html", {"form": form})
 
 
+def onboarding(request):
+    # TODO uncomment this and make else redirect to page not found or home
+    # if(request.session.get("onboarding")):
+    return render(
+        request,
+        "accounts/onboarding.html",
+    )
+
+
+def onboarding_genre_view(request):
+    """
+    This function handles the POST/GET requests for the onboarding genre form.
+    """
+    if not request.user.is_authenticated:
+        return redirect("accounts:login")
+
+    if request.method == "POST":
+        form = GenreForm(request.POST)
+        if form.is_valid():
+            liked_genres = []
+            disliked_genres = []
+            blocked_genres = []
+            for genre, preference in form.cleaned_data.items():
+                if preference == "like":
+                    liked_genres.append(genre)
+                elif preference == "dislike":
+                    disliked_genres.append(genre)
+                elif preference == "block":
+                    blocked_genres.append(genre)
+            with transaction.atomic():
+                request.user.liked_genres = liked_genres
+                request.user.disliked_genres = disliked_genres
+                request.user.excluded_genres = blocked_genres
+                request.user.save()
+            return redirect("/accounts/onboarding/movies")
+    else:
+        # Fetch the user's preferences
+        liked_genres = request.user.liked_genres
+        disliked_genres = request.user.disliked_genres
+        blocked_genres = request.user.excluded_genres
+
+        # Prepare initial data dictionary for the form
+        initial_data = {}
+        for genre in GENRES:
+            genre_value = genre[0]
+            if genre_value in liked_genres:
+                initial_data[genre_value] = "like"
+            elif genre_value in disliked_genres:
+                initial_data[genre_value] = "dislike"
+            elif genre_value in blocked_genres:
+                initial_data[genre_value] = "block"
+        form = GenreForm(initial_preferences=initial_data)
+    return render(
+        request,
+        "accounts/onboarding_genres.html",
+        {
+            "form": form,
+        },
+    )
+
+
+def onboarding_movie_view(request):
+    """
+    This function handles the POST/GET requests for the onboarding movie form.
+    POST saves the user's preferences while GET renders the page.
+    """
+    if request.method == "POST":
```

Comment:
always go the path of least nesting

if request.method != "POST":
  return render

then the post related stuff

OR

if request.method == "GET":
  return render